libbeat/processors/actions/extract_field.go

```diff
@@ -0,0 +1,91 @@
+package actions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type extract_field struct {
+	Field     string
+	Separator string
+	Index     int
+	Target    string
+}
+
+/*
+This one won't be registered (yet)
+
+func init() {
+	processors.RegisterPlugin("extract_field",
+		configChecked(NewExtractField,
+			requireFields("field", "separator", "index", "target"),
+			allowedFields("field", "separator", "index", "target", "when")))
+}
+*/
+
+func NewExtractField(c common.Config) (processors.Processor, error) {
+	config := struct {
+		Field     string `config:"field"`
+		Separator string `config:"separator"`
+		Index     int    `config:"index"`
+		Target    string `config:"target"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the extract_field configuration: %s", err)
+	}
+
+	/* remove read only fields */
+	for _, readOnly := range processors.MandatoryExportedFields {
+		if config.Field == readOnly {
+			return nil, fmt.Errorf("%s is a read only field, cannot override", readOnly)
+		}
+	}
+
+	f := extract_field{
+		Field:     config.Field,
+		Separator: config.Separator,
+		Index:     config.Index,
+		Target:    config.Target,
+	}
+	return f, nil
+}
+
+func (f extract_field) Run(event common.MapStr) (common.MapStr, error) {
+	fieldValue, err := event.GetValue(f.Field)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting field '%s' from event", f.Field)
+	}
+
+	value, ok := fieldValue.(string)
+	if !ok {
+		return nil, fmt.Errorf("Could not get a string from field '%s'", f.Field)
+	}
+
+	parts := strings.Split(value, f.Separator)
+	parts = deleteEmpty(parts)
```

Comment:
I don't fully get this question, but will explain why I had to delete empty parts. When you split a string like `/var/lib/docker/containers/foo` you get `[ '', 'var', 'docker', 'containers', 'foo' ]`. It's counter intuitive to ask for index 4 and get `containers` (I would expect `foo`) because of the empty string at the beginning. Things get worse if you have things like `/var//lib/...`